### PR TITLE
Extend the table fetcher to handle DT tables too

### PR DESCRIPTION
**Pull request by vedhav**
Extend the `get_active_module_tws_output` to also work for `DT::DTOutput` which is now renamed to `get_active_module_table_output` along with `get_active_module_plot_output`

Example to test:

```r
devtools::load_all("../teal")
library(teal.modules.clinical)

data <- teal_data()
data <- within(data, {
  ADSL <- teal.data::rADSL
  ADLB <- teal.data::rADLB
})

datanames <- c("ADSL", "ADLB")
teal.data::datanames(data) <- datanames
teal.data::join_keys(data) <- teal.data::default_cdisc_join_keys[datanames]

app_driver <- TealAppDriver$new(
  data = data,
  modules = tm_t_pp_laboratory(
    label = "Vitals",
    dataname = "ADLB",
    patient_col = "USUBJID",
    paramcd = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "PARAMCD"),
      selected = "PARAMCD"
    ),
    param = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "PARAM"),
      selected = "PARAM"
    ),
    timepoints = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "ADY"),
      selected = "ADY"
    ),
    anrind = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "ANRIND"),
      selected = "ANRIND"
    ),
    aval_var = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "AVAL"),
      selected = "AVAL"
    ),
    avalu_var = teal.transform::choices_selected(
      choices = teal.transform::variable_choices(data[["ADLB"]], "AVALU"),
      selected = "AVALU"
    )
  )
)

app_driver$get_active_module_table_output("lab_values_table", 2)
```

### CODE REVIEW

File: R/TealAppDriver.R [hunk 1]
```diff
@@ -253,13 +253,14 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
...
       checkmate::check_number(which, lower = 1)
       checkmate::check_string(tws)
-      table <-
-        rvest::html_table(
-          self$get_html_rvest(
-            self$active_module_element(sprintf("%s-table-with-settings", tws))
-          ),
-          fill = TRUE
-        )
+      table_selector <- ifelse(
```
Comment (kartikeyakirar) [inline R/TealAppDriver.R:256]: what if we have both table in module ?

**Review by gogonzo — COMMENTED** [review]
Here we are, decision about including method to `TealAppDriver` has already some consequences. 

Name `get_active_module_tws_output` is not accurate anymore, but what is worse is that function doesn't handle single tws case but has hardcoded condition for not-tws. 


Probably it should be called like this from the beginning. 

```r
app_driver$get_active_module_table_output(
  table_id = "my_table-table-with-settings" 
)
```

File: R/TealAppDriver.R (hunk 1 above)
Comment (vedhav) [reply R/TealAppDriver.R:256]: I did not see cases like this in all the 36 modules in tmc.

Comment (vedhav) [reply R/TealAppDriver.R:256]: But, I see your point, I will make this as an argument to make the selection.

**Comment by vedhav** [comment]
Yeah, true. This should be `get_active_module_table_output`, And, it's companion `get_active_module_plot_output` which will slowly extend to other plot types if observed.

File: R/TealAppDriver.R (hunk 1 above)
Comment (vedhav) [reply R/TealAppDriver.R:258]: Done in b36be57b69269c7703ed2843ce7cc84eba087996

File: R/TealAppDriver.R [hunk 2]
```diff
@@ -273,7 +276,7 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
     #' @param pws (`character(1)`) `plot_with_settings` namespace name.
     #'
     #' @return The `src` attribute as `character(1)` vector.
-    get_active_module_pws_output = function(pws) {
+    get_active_module_plot_output = function(pws) {
```
Comment (kartikeyakirar) [inline R/TealAppDriver.R:279]: same here as well https://github.com/insightsengineering/teal/pull/1210/files#r1582990384

File: R/TealAppDriver.R [hunk 3]
```diff
@@ -243,23 +243,26 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
...
     #' This function will only access outputs from the name space of the current active teal module.
     #'
     #' @param tws (`character(1)`) `table_with_settings` namespace name.
     #' @param which (integer) If there is more than one table, which should be extracted.
+    #' @param is_dt_table (logical) If the table is a `DT::DTOutput`.
+    #' By default it will look for  a table that is built using `teal.widgets::table_with_settings`.
     #'
     #' @return The data.frame with table contents.
-    get_active_module_tws_output = function(tws, which = 1) {
+    get_active_module_table_output = function(tws, which = 1, is_dt_table = FALSE) {
```
[the remark this answers is not in the document]
Comment (vedhav) [reply R/TealAppDriver.R:255]: True, `table_id` makes more sense here.

File: R/TealAppDriver.R (hunk 2 above)
Comment (vedhav) [reply R/TealAppDriver.R]: Done

**Comment by gogonzo** [comment]
> ```r
> app_driver$get_active_module_table_output(
>   table_id = "my_table-table-with-settings" 
> )
> ```

Function name changed so we'll need to update test everywhere downstream. Why not to apply my suggestion . This should be relatively straightfoward. 

```r
app_driver$get_active_module_table_output(
  table_id = "my_table-table-with-settings" 
)
```


While `"my_table-table-with-settings"` theoreticaly could be handled by `ns_pws` (same as `pws_des_input`) but it is simple enoygh that I could write it manually.

**Comment by vedhav** [comment]
I am also okay with this change. But, I would still prefer to have the `table-with-settings` in once place for two reasons:
1. All the input ids used in our teal module tests can be directly found in the module code. So, I would like to keep it that way if possible. As, it is easy to search for "table" in the code with `teal.widgets::table_with_settings_ui(ns("table"))`.
2. Changing this class in the future will be easy if it is in one place.

**Comment by vedhav** [comment]
Okay, I already see that some of the ids are implemented as you suggested. So, I'm onboard. I'll remove this logic from there.

**Review by kartikeyakirar — APPROVED** [review]
LGTM

**Pull request merged**